tests/test_restartability.py

```diff
@@ -54,3 +56,26 @@ def test_faker_dates_work(self, write_row):
         continuation_file = StringIO()
         generate(StringIO(yaml), generate_continuation_file=continuation_file)
         assert "a_date" in continuation_file.getvalue()
+
+    @mock.patch("snowfakery.output_streams.DebugOutputStream.write_row")
+    def test_circular_references(self, write_row):
+        yaml_data = """
+            - object: parent
+              nickname: TheParent
+              fields:
+                child:
+                    - object: child
+                      nickname: TheChild
+                      fields:
+                        parent:
+                            reference:
+                                TheParent
+            """
+        continuation_file = StringIO()
+        generate(StringIO(yaml_data), generate_continuation_file=continuation_file)
+        continuation_yaml = continuation_file.getvalue()
+        print(continuation_yaml)
+        generate(
+            StringIO(yaml_data),
+            continuation_file=StringIO(continuation_yaml),
+        )
```

Comment:
@prescod There's no PR description, and the test makes no assertion. What is the change in behavior that you're implementing here?

Reply:
@prescod thanks for adding a PR description but I'm still not clear: does this silently leave data out of the continuation file, or does it somehow work even though those references are not included? It would be helpful to have an assertion in this test about what data was generated.

Reply:
I had also added a comment to the code. Does it help?

"""    Do not include related ObjectRows because circular
        references in serialization formats cause problems."""

The gist is that if object A refers to object B, object A will be serialized but B will not (unless it independently would have been). This prevents A->B->A loops.

I don't know of a use-case where maintaining the A->B relationship across continuations is useful. If you need access to B across continuations you can give it a nickname.

Reply:
> I don't know of a use-case where maintaining the A->B relationship across continuations is useful.

This is the piece I was looking for confirmation of.